### ISSUE TITLE
verify that a set of labels is overriden in every subsequent layer

### ISSUE THREAD

**Issue by TomasTomecek** [user-cont/colin#146]
@lachmanfrantisek you were right, it's possible to do this. I just realized that osbs squashes all layers. Therefore we should be able to:

1. detect if a label from the set is defined
2. check if it's overriden in every subsequent layer

Optimization detail: try as hard to do this via API without doing `docker save` -- that's one very expensive operation -- we should be able to accomplish this just by doing inspect on all layers.

Just got great pointers from @lcarva:

    The only difference is for base images (rhel), because they will have 2 layers. 
    We do store Dockefiles for each image in hierarchy under /root/build-info if that's useful.

**Comment by lachmanfrantisek** [comment]
First idea, how to retrieve such info:

- Dockerfile -> `FROM` -> Image
- Container ->  `Image` -> Image
- Image:
    - `Parent` (usually not set)
    - `['RootFS']['Layers']` --> search in all images (not pulled images are not present, only their layers)
    - `/root/buildinfo` directory with dockerfiles (not a general rule)

@TomasTomecek  WDYT, are there any other options?

**Comment by TomasTomecek** [comment]
['RootFS']['Layers'] --> search in all images (not pulled images are not present, only their layers)

do the layers contain correct metadata?

    /root/buildinfo directory with dockerfiles (not a general rule)

probably our 3rd option

**Comment by lachmanfrantisek** [comment]
> do the layers contain correct metadata?

I am not sure, but I think there is no metadata about an image in the layer and you have to pull the base image to get the metadata of the base...

**Comment by lachmanfrantisek** [comment]
If I try this:

```python
for i in d.images.list(all=True):
    if "sha256:b1e3561f65d.....10" in i.attrs["RootFS"]["Layers"]:
        print(i)
```

I will get the parent object only when I pulled them explicitly...

**Comment by TomasTomecek** [comment]
> I am not sure, but I think there is no metadata about an image in the layer and you have to pull the base image to get the metadata of the base...

You're correct. It's not possible to inspect them.

**Comment by lslebodn** [comment]
This issue fixed in pr #230
